### ISSUE TITLE
problem installing

### ISSUE THREAD

**Issue by dimis40** [rg3/youtube-dl#7397]
I am trying to install it on mac os x using curl and i receive this error code

(23) Failed writing body (0 != 4096)

**Comment by dstftw** [comment]
Post the **full** output.
Alternatively use homebrew.

**Comment by gaming-hacker** [comment]
what version of curl? and where is it?  the OS X default is generally outdated.
which curl
curl --version

don't use php from homebrew use the one from here, better maintained with more extensions pre-built
http://php-osx.liip.ch

**Comment by dimis40** [comment]
I use the curl version

curl 7.37.1

2015-11-09 23:55 GMT+02:00 G Alexander <notifications@github.com>:

**Comment by gaming-hacker** [comment]
where is it?
at bash prompt > which curl

stop deleting my comments or i'll refuse to help or push my changes back to your repo.

**Comment by dstftw** [comment]
Stop posting nonsense and unrelated stuff.

**Comment by gaming-hacker** [comment]
how is diagnosing curl unrelated?   no more help from then, i'll let people grab my fork and help them even if not related to code.

**Comment by dstftw** [comment]
I'm about 3rdparty php.
I'm closing this issue since no requested info was provided. As I've already mentioned you can use homebrew to install youtube-dl just fine.

**Comment by dimis40** [comment]
ok i will use homebrew

thanks

2015-11-10 18:53 GMT+02:00 Sergey M. <notifications@github.com>:

**Comment by aminozuur** [comment]
I am trying to reinstall youtube-dl after a fresh OS X Sierra install and I am running into the same 'failed writing error' problem.

**Comment by Hrxn** [comment]
When (or where) exactly does this problem occur? 
Here? `sudo curl -L https://yt-dl.org/downloads/latest/youtube-dl -o /usr/local/bin/youtube-dl`

If I had to guess, OS X / macOS has some security mechanism to prevent software from 'untrusted' sources, I think they call it Gatekeeper or something..

**Comment by aminozuur** [comment]
I just disabled Gatekeeper (in Sierra you'll have to use this [terminal command](http://apple.stackexchange.com/questions/243687/allow-applications-downloaded-from-anywhere-in-macos-sierra)) to allow all apps. But after that, installing youtube-dl using the above command still does not work :(. 

I'm still getting the same error:
curl: (23) Failed writing body (0 != 16360)

**Comment by yan12125** [comment]
Post the output of the following command:
```sudo curl -v -L https://yt-dl.org/downloads/latest/youtube-dl -o /usr/local/bin/youtube-dl```
And please also check whether root has write access to ```/usr/local/bin/``` or not.

**Comment by aminozuur** [comment]
Yan1215, I just did.
TL;DR: it returns the 'Failed writing body' error once again..

Here is the full paste:
http://pastebin.com/3bym48pf

I am doing this from from my main admin account, so I guess it has write access.

**Comment by yan12125** [comment]
From your log I guess there's no ```/usr/local/bin/```. Could you create it manually and try again?
```
sudo mkdir -p /usr/local/bin/
```
You may want to add ```/usr/local/bin/``` to ```$PATH```, too.

**Comment by aminozuur** [comment]
I think I created the directory. Although, when I run `/usr/local/bin/youtube-dl`, it returns `No such file or directory`. hmm.


I still continued and added the directory to $PATH. When I run 'echo $PATH' it now returns:
`echo $PATH
/usr/local/bin/:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/bin/`

But when I try to run youtube-dl, it now says `Permission denied`.

I understand these these errors are the results of my own ignorance regarding CLI #damn

**Comment by yan12125** [comment]
Could you try: ```python /usr/local/bin/youtube-dl```
